src/widget/form/settings/advancedform.cpp

```diff
@@ -30,16 +40,32 @@ AdvancedForm::AdvancedForm() :
     bodyUI = new Ui::AdvancedSettings;
     bodyUI->setupUi(this);
 
+    Settings &s = Settings::getInstance();
+    bodyUI->cbEnableIPv6->setChecked(s.getEnableIPv6());
     bodyUI->cbMakeToxPortable->setChecked(Settings::getInstance().getMakeToxPortable());
+    bodyUI->cbEnableUDP->setChecked(!s.getForceTCP());
+    bodyUI->proxyAddr->setText(s.getProxyAddr());
+    int port = s.getProxyPort();
+    if (port != -1)
+        bodyUI->proxyPort->setValue(port);
 
-    connect(bodyUI->cbMakeToxPortable, &QCheckBox::stateChanged, this, &AdvancedForm::onMakeToxPortableUpdated);
-    connect(bodyUI->resetButton, SIGNAL(clicked()), this, SLOT(resetToDefault()));
-
-    for (QCheckBox *cb : findChildren<QCheckBox*>()) // this one is to allow scrolling on checkboxes
-    {
-        cb->installEventFilter(this);
-    }
+    bodyUI->proxyType->setCurrentIndex(static_cast<int>(s.getProxyType()));
+    onUseProxyUpdated();
 
+    // portable
+    connect(bodyUI->cbMakeToxPortable, &QCheckBox::stateChanged, this, &AdvancedForm::onMakeToxPortableUpdated);
+    connect(bodyUI->resetButton, &QPushButton::clicked, this, &AdvancedForm::resetToDefault);
+    //connection
+    void (QComboBox::* currentIndexChanged)(int) = &QComboBox::currentIndexChanged;
+    void (QSpinBox::* valueChanged)(int) = &QSpinBox::valueChanged;
+    connect(bodyUI->cbEnableIPv6, &QCheckBox::stateChanged, this, &AdvancedForm::onEnableIPv6Updated);
+    connect(bodyUI->cbEnableUDP, &QCheckBox::stateChanged, this, &AdvancedForm::onUDPUpdated);
+    connect(bodyUI->proxyType, currentIndexChanged, this, &AdvancedForm::onUseProxyUpdated);
+    connect(bodyUI->proxyAddr, &QLineEdit::editingFinished, this, &AdvancedForm::onProxyAddrEdited);
+    connect(bodyUI->proxyPort, valueChanged, this, &AdvancedForm::onProxyPortEdited);
+    connect(bodyUI->reconnectButton, &QPushButton::clicked, this, &AdvancedForm::onReconnectClicked);
```

Comment:
Yes, I just implemented the `qTox::RecursiveSignalBlocker` few days ago. I'll make it available in master, so you don't have to worry about that. Only thing you'll need to do is to rebase on master. I'll let you know, when it is merged.